rfcs/0027-formalize-documentation-organization.md

```diff
@@ -0,0 +1,442 @@
+beginner something they need to know. This is in contrast with a how-to, which
+is meant to be an answer to a question that a user with some experience would
+ask.
+
+Tutorials need to be repeatable and reliable, because the lack of success means
+a user will look for other solutions.
+
+### How-to Guides
+
+These are step by step guides on how to solve particular problems. The user can
+ask meaningful questions, and the documents provide answers. An examples of
+this type of document might be, “how do I compile an optimized model for ARM
+architecture?” or “how do I compile and optimize a TensorFlow model?” These
+documents should be open enough that a user could see how to apply it to a new
+use case. Practical usability is more important than completeness. The title
+should tell the user what problem the how-to is solving.
+
+How are tutorials different from how-tos? A tutorial is oriented towards the
+new developer, and focuses on successfully introducing them to the software and
+community. A how-to in contrast focuses on accomplishing a specific task within
+the context of basic understanding. A tutorial helps to onboard, a how-to helps
+to accomplish a task.
+
+### Reference
+
+Reference documentation describes how the software is configured and operated.
+APIs, key functions, commands, and interfaces are all candidates for reference
+documentation. These are the technical manuals that let users build their own
+interfaces and programs. They are information oriented, focused on lists and
+descriptions. You can assume that the audience has a grasp on how the software
+works and is looking for specific answers to specific questions. Ideally, the
+reference documentation should have the same structure as the code base and
+generated automatically as much as possible.
+
+### Explanations (Deep Dive)
+
+Background material on a topic. These documents help to illuminate and
+understand the application environment. Why are things the way they are? What
+were the design decisions, what alternatives were considered, what are the RFCs
+describing the existing system. This includes academic papers and links to
+publications relevant to the software. Within these documents you can explore
+contradictory and conflicting position, and help the reader make sense of how
+and why the software was built the way it is. It’s not the place for how-to’s
+and technical descriptions, and instead focuses on higher level concepts.
+
+## Special considerations for TVM
+
+The TVM comunity has some special considerations that require deviation from
+the simple docs style outlined by Divio. The first is that there is frequently
+overlap between the user and developer communities. Many projects document the
+develper and user experience with separate systems, but it is appropriate to
+consider both in this system, with differentiations where appropriate. As a
+result the tutorials and how-tos will be divided between "User Guides" and
+"How-to Guides".
```

Comment:
User guides are for users and how-to guides are for developers? Maybe state that explicitly?